crates/ruma-common/src/events/room/member.rs

```diff
@@ -269,6 +288,15 @@ impl SignedContent {
     }
 }
 
+/// The details of a member event required to calculate a [`MembershipChange`].
+pub struct MembershipDetails<'a> {
+    // These fields are intentionally private, this struct is only meant for
```

Comment:
I don't think there's a good reason to keep those private since we do have pub methods returning the struct.